airflow/hooks/dbapi.py

```diff
@@ -176,18 +178,32 @@ def run(self, sql, autocommit=False, parameters=None, handler=None):
         :param handler: The result handler which is called with the result of each statement.
         :type handler: callable
         :return: query results if handler was provided.
+        :param split_statements: If true, split sql statements
+        :type split_statements: bool
         """
         scalar = isinstance(sql, str)
         if scalar:
             sql = [sql]
+        sql_statements = sql
+
+        if split_statements:
+            sql_statements = []
+            for q in sql:
+                sql_statements.append(
+                    [
+                        s[:-1] if s[-1] == ';' else s
```

Comment:
Do we need to strip `;`? I know most SQL engines don’t really care about them.

If we do, this is probably better

```suggestion
                        s.rstrip(";")
```

Reply:
@uranusjr Agree, I saw that in drill and presto, I assumed that it's neccesary, but as you said, most sql engine handles ;